contrib/sssd.spec.in

```diff
@@ -1047,7 +1047,7 @@ install -D -p -m 0644 contrib/sssd.sysusers %{buildroot}%{_sysusersdir}/sssd.con
 %if %{use_sssd_user}
 %pre common
 %if %{use_sysusers}
-%sysusers_create_compat contrib/sssd.sysusers
+%sysusers_create_compat %{buildroot}%{_sysusersdir}/sssd.conf
```

Comment:
s/%{buildroot}//

Reply:
Hi,

but if you remove `%{buildroot}` you will end up with `/usr/lib/sysusers.d/sssd.conf`, i.e. referencing a file from the filesystem of the build host.

The `%sysuser_create_compat` macro is executed at the time the scriptlets for the rpm packages are created and it calls `/usr/lib/rpm/sysusers.generate-pre.sh` with the sysusers config file as argument and the output
```
# generated from sssd.sysusers
getent group 'sssd' >/dev/null || groupadd -r 'sssd' || :
getent passwd 'sssd' >/dev/null || \
    useradd -r -g 'sssd' -d '/' -s '/sbin/nologin' -c 'User for sssd' 'sssd' || :
```
is written into the scriptlet.

Since we install the sysusers config file as well in `/usr/lib/sysusers.d/` the `%systemd_post` macro will create users and groups if then not exists as well. But since we need the user and group so that the files installed by rpm will have proper ownership doing this in `%post` is too late. And we cannot let systemd do it at install time in `%pre` because the sysusers config file is not installed at this time.

To cut it short, I still see no other way than having the sysusers config file as additional source file.

bye,
Sumit

Reply:
Hmm... you are right that %pre is executed before the file is actually installed.

But what difference does it make if we read it from out tarball or from an additional source file? The macro is evaluated before our tarball is extracted?

Reply:
At this point I wonder if `sysusers_create_compat` is really better than "manual" user/group creation...

Reply:
> Hmm... you are right that %pre is executed before the file is actually installed.
> 
> But what difference does it make if we read it from out tarball or from an additional source file? The macro is evaluated before our tarball is extracted?

Hi,

no, it is evaluated after the tar ball is extracted, after SSSD is build and after content of the BUILD and BUILDROOT directories is removed (I put an `ls` in the `pre` section to check this), so there is no extracted or built data available anymore.

HTH

bye,
Sumit

Reply:
Looks like 736430aa0ed0f9c9e36315ea97de65908c29f590 needs to be reverted fully, including sssd.sysusers creation by autoconf?

I don't understand why but otherwise build fails with:
```
error: Bad file: /builddir/build/SOURCES/sssd.sysusers: No such file or directory
    Bad file: /builddir/build/SOURCES/sssd.sysusers: No such file or directory
```